cmd/slinky/main.go

```diff
@@ -66,10 +69,24 @@ var (
 
 func init() {
 	rootCmd.Flags().StringVarP(
-		&oracleCfgPath,
+		&marketMapProvider,
+		"marketmap-provider",
+		"",
+		marketmap.Name,
+		"MarketMap provider to use (marketmap_api, dydx_api, dydx_research_api).",
+	)
+	rootCmd.Flags().StringVarP(
+		&legacyOracleCfgPath,
 		"oracle-config-path",
 		"",
-		"oracle.json",
+		"",
```

Comment:
Changing the name of the flag will fully break people and even if we keep the flag name the same, changing the default will probably break some people. I think it's fine as long as we call everything out clearly in our docs. We can maybe have a section which details `0.4.X` config and its deprecation/migration path going forward. 

It seems like we want to roll this into `1.0.0` as a breaking change and maybe fully remove the old config in `1.1.X`? Let me know how you're thinking about this as a user experience.

Reply:
We probably want to use this https://pkg.go.dev/github.com/spf13/pflag#FlagSet.MarkDeprecated